### DISCUSSION TITLE
Convert a Tasmota Rule into Berry Script

### DISCUSSION THREAD

**Discussion by Salvoitaly2023** [arendst/Tasmota#22366] [Q&A]
Hello community,
I have a problem.
I would like to convert a Tasmota Rule into Berry Script.

This is the rule:

Rule1 on AM2301#Temperature do DisplayText [f0s1l3c1]Temperatur: %value% ~f8C ~20 endon 
on AM2301#Humidity do DisplayText [f0s1l5c1]Feuchtigkeit: %value% %H ~20 endon 
on AM2301#DewPoint do DisplayText [f0s1l7c1]Taupunkt: %value% ~f8C ~20 endon 
on Counter#C1 do backlog var1 %value%; MULT1 0.01; DisplayText [f0s1l1c1]GAS: %var1% m3 ~20 endon

Who can please give me some help on how to convert this? I've been searching the Internet for days for solutions, but unfortunately I haven't found anything suitable.

Thank you very much

**Comment by sfromis** [comment]
One approach could be like this: (untested)
```
tasmota.remove_rule('AM2301')
tasmota.add_rule('AM2301',
  def (values)
    var temperature = values['Temperature']
    var humidity = values['Humidity']
    var dewpoint = values['DewPoint']
    tasmota.cmd(f'DisplayText '
      '[f0s1l3c1]Temperatur: {temperature} ~f8C ~20'
      '[f0s1l5c1]Feuchtigkeit: {humidity} %H ~20'
      '[f0s1l7c1]Taupunkt: {dewpoint} ~f8C ~20'
    )
  end)
```
One `DisplayText` command instead of 3 should work as well. And another `tasmota.add_rule` for the counter, where you can divide by 100 when you pick the count from the input, instead of that `var1` usage.

BTW. Your old rule does not really work, as the `%var1%` in the `DisplayText` command is from the previous invocation of the rule, instead of the just calculated value.

You could also do it in a very different way using HASPmota for display styling, without needing any Berry program as intermediate processing.

**Comment by Salvoitaly2023** [comment]
It works perfectly! Thank you very much!
